conditional_analysis.py

```diff
@@ -0,0 +1,423 @@
+                if to_path(f'{test_output_path}.set').exists():
+                    continue
+
+                # define gene-specific null output (rda + varianceRatio)
+                null_path = (
+                    f'{fit_null_files_path}/{celltype}/{chromosome}/{celltype}_{gene}'
+                )
+                # define gene-specific cis window or group file
+                if common_or_rare == 'common':
+                    cis_window_or_group_file = f'{cis_window_or_group_files_path_chrom}/{gene}_{cis_window_size}bp.tsv'
+                elif common_or_rare == 'rare':
+                    cis_window_or_group_file = f'{cis_window_or_group_files_path_chrom}/{gene}_{cis_window_size}bp{group_file_specs}.tsv'
+
+                step2_output = build_run_conditional_analysis_command(
+                    job=step2_job,
+                    test_key=test_key,
+                    test_output_path=test_output_path,
+                    vcf_group=vcf_group,
+                    chrom=(chromosome[3:]),
+                    cis_window_or_group_file=cis_window_or_group_file,
+                    gmmat_model_path=f'{null_path}.rda',
+                    variance_ratio_path=f'{null_path}.varianceRatio.txt',
+                    common_or_rare=common_or_rare,
+                    conditional_string=conditional_string,
+                )
+                jobs_in_vm += 1
+                if common_or_rare == 'common':
+                    # step 3 (gene-level p-values)
+                    saige_gene_pval_output_file = output_path(
+                        f'{celltype}/{chromosome}/{celltype}_{gene}_cis_gene_pval'
+                    )
+                    if not to_path(saige_gene_pval_output_file).exists():
+                        job3 = build_obtain_gene_level_pvals_command(
+                            gene_name=gene,
+                            saige_sv_output_file=step2_output,
+                            saige_gene_pval_output_file=output_path(
+                                f'{celltype}/{chromosome}/{celltype}_{gene}_cis_gene_pval'
+                            ),
+                        )
+                        job3.depends_on(step2_job)
+                        # add this job to the list of jobs for this cell type
+                        celltype_jobs.setdefault(celltype, []).append(job3)
+
+            if jobs_in_vm >= jobs_per_vm:
+                step2_job = create_second_job(vcf_file_path)
+                jobs_in_vm = 0
+    # summarise results (per cell type)
+    for celltype in celltypes:
+        logging.info(f'start summarising results for {celltype}')
+        summary_output_path = (
+            f'summary_stats/{celltype}_all_cis_{common_or_rare}_gene_level_results.tsv'
+        )
+
+        summarise_job = get_batch().new_python_job(
+            f'Summarise {common_or_rare} results for {celltype}'
+        )
+        summarise_job.depends_on(*celltype_jobs[celltype])
```

Comment:
```suggestion
        if celltype in celltype_jobs:
            summarise_job.depends_on(*celltype_jobs[celltype])
```

my guess is that by running it in fragments you didn't have any previous jobs in this batch. In a full end to end run you'd expect to find all the jobs, but not if the previous outputs exist when you set off this run.